side-products/merkl/integration-guide.md

```diff
@@ -20,65 +22,79 @@ The `pools` object also has for each incentivized pool details about the differe
 
 You can filter from the \`pools\` object to only display the pools of your choice (for instance pools with tokens in a subset and with an active distribution).
 
-## Display user rewards
+## Displaying user rewards
 
 The API lets you get for a user through the `rewardsPerToken` object how much of rewards can be claimed for a specific pool.
 
 This object maps each reward token to the total amount of rewards accumulated on the pool (it is also inclusive of the rewards that have already been claimed) as well as to the amount of unclaimed tokens on the pool. It also details where these tokens have been earned (Uniswap V3, Gamma, ...).
 
-## Claim user rewards
+## Claiming user rewards
 
-On top of displaying how much has been earned by a specific address, the API can also be used to build the payload of a claim transaction of a user which accumulated rewards.&#x20;
+On top of displaying how much has been earned by a specific address, the API can also be used to build the payload of a claim transaction of a user which accumulated rewards.
 
-When called for a specific user, it returns in a `transactionData` payload the list of tokens with for each a Merkle proof to use to claim the associated rewards.
+When called for a specific user, it returns in a `transactionData` payload with the list of `tokens` to claim, the `amounts`, and a Merkle `proof` required to claim each of them.
 
-Rewards are claimable per token: meaning that if you have accumulated rewards of several tokens, you may choose to only claim your rewards of one token type, but you may also choose to claim all your token rewards at once.
+_Rewards are claimable per token: meaning that if you have accumulated rewards of several tokens, you may choose to only claim your rewards of one token type, but you may also choose to claim all your token rewards at once._
 
 The contract on which rewards should be claimed is the `Distributor` contract which address can be found on [this docs](helpers.md#🧑‍💻-smart-contracts), or on the [Angle SDK](https://github.com/AngleProtocol/sdk).
 
+You have multiple options to do that:
+
+- Rely on Angle's API: we build the claim transaction payload for you and the associated proof, and you just call our API. This is the example shown below.
+- Build the proof yourself and join it to the transaction data from the API. You can find a Github repository below showing how to do that.
```

Comment:
Not sure to understand the difference between the two cases -> in the API we're never sending the contract address I think nor building the transaction for you (cf the script below)

Reply:
Isn't this fetching the contract address from the SDK ? 

<img width="625" alt="CleanShot 2023-04-21 at 15 18 40@2x" src="https://user-images.githubusercontent.com/33679821/233645984-174476d2-ba2c-4474-ac28-cdb937d24903.png">

Reply:
Yes but it does not change anything from the script like -> if you don't update the SDK, address will still be considered valid